src/Functions/transform.cpp

```diff
@@ -163,7 +163,16 @@ namespace
 
             ColumnPtr default_non_const;
             if (!cache.default_column && arguments.size() == 4)
+            {
                 default_non_const = castColumn(arguments[3], result_type);
+                if (in->size() > default_non_const->size())
+                {
+                    throw Exception(
```

Comment:
If columns in a blog are of different sizes, it is LOGICAL_ERROR, not BAD_ARGUMENTS (because a user cannot make it).